README.md
![alt tag](https://bitbucket.org/hrznstudio/mo-legacy-edition/raw/1.12.2/Banner.png)

[![Discord](https://img.shields.io/discord/422424112863117312.svg?style=for-the-badge)](https://discord.gg/758eCD7)

## Table of Contents
* [About](#about)
* [Features](#features)
* [Contacts](#contacts)
* [Issues](#issues)
* [Building](#building)

## About
Matter Overdrive is a Minecraft mod inspired by the popular Sci-fi TV series Star Trek. It dwells in the concept of replicating and transforming one type matter into another.
Although it may seem overpowered, Matter Overdrive takes a more realistic approach and requires the player to build a complex system before even the simplest replication can be achieved.

## Contacts
* [Email](mailto:contact@hrznstudio.com)
* [Website](https://hrzn.studio/mo)
* [Discord](https://discord.gg/758eCD7)

## Features
* [Matter Scanner](https://mo.simeonradivoev.com/items/matter_scanner/), for scanning matter patterns for replication.
* [Replicator](https://mo.simeonradivoev.com/items/replicator/), for transforming materials.
* [Decomposer](https://mo.simeonradivoev.com/items/decomposer/), for breaking down materials to basic form.
* [Transporter](https://mo.simeonradivoev.com/items/transporter/), for beaming up.
* [Phaser](https://mo.simeonradivoev.com/items/phaser/), to set on stun.
* [Fusion Reactors](https://mo.simeonradivoev.com/fusion-reactor/) and [Gravitational Anomaly](https://mo.simeonradivoev.com/items/gravitational_anomaly/)
* Complex Networking for replication control.
* Star Maps, with Galaxies, Stars and Planets
* [Androids](https://mo.simeonradivoev.com/android-guide/), become an Android and learn powerful RPG like abilities, such as Teleportation and Forefield Shields.


![Matter Overdrive Blocks and Items](https://media-elerium.cursecdn.com/attachments/210/237/main_screenshot.png)

Mini-Changelog (from VanMuur):

1. Turned Pattern Storage 180 degrees when placing so the front is facing you, like all other machines.
2. Updated all machines to accept the Matter Network Cable from any side, not just the back.

Issues:

1. Attempted to get matter storage pipes to attach to more than just the back of the replicator. It connects, but nothing transfers.

Todo:

1. Allow energy transfer across matter network cables. That way connectivity and power go over one set of cables.
2. Make the microwave able to "cook" food like furnaces do, getting power from RF. If ores are put in the microwave, make it explode.
3. Fix network flash drives to properly filter destinations.

Unreleased:

1. Created the TileEntity for the Microwave in preparation for getting it to work. It still doesn't have proper functionality yet.



CHANGELOG.md
MatterOverdrive-FHE - Changelog
===============================

Unreleased
----------
* Multiple fixes and debugging to try to get the transporter's energy bar to show properly. (VanMuur)

v0.1.8.1 (2020-11-08 07:45:01 -0500)
------------------------------------
* Updating version numbers for release. (VanMuur)
* Made energy per unit travelled used configurable. Same with matter plasma. Added tooltips for energy and matter available in the trnasporter. Attempted to fix energy displayed bug. (VanMuur)

v0.1.8 (2020-11-07 10:44:00 -0500)
----------------------------------
* Updating version number for release. (VanMuur)
* Fixing matter transporter bugs, and disabled multiple container actions. (VanMuur)

v0.1.7.6 (2020-11-07 09:18:03 -0500)
------------------------------------
* Fixing version number issue with 0.1.7.6. (VanMuur)
* Updated version number. (VanMuur)
* Fixed the decomposer so it won't display processing if its energy is zero. Matter containers stack split is still bugged. (VanMuur)
* Changed recipe (Gandalf360one)
* Changed recipe (Gandalf360one)
* changed recdipe (Gandalf360one)

v0.1.7.5 (2020-11-06 18:06:35 -0500)
------------------------------------
* Removed a log file put in the wrong directory. (VanMuur)
* Fixed matter transporters so they didn't teleport to the wrong location. (VanMuur)

v0.1.7.4 (2020-11-06 17:46:34 -0500)
------------------------------------
* Fixed a small bug where if your inventory was full, it wouldn't dropped a filled matter container properly. (VanMuur)
* Changed to facing for the smoke (Gandalf360one)

v0.1.7.3 (2020-10-31 14:33:27 -0400)
------------------------------------
* Updated version for bug fixes. (VanMuur)
* Added recipes for the wide and small holo sights. (VanMuur)
* Fixed matter containers to split on a stack being filled. Also added recipes for the holo sights. (VanMuur)

v0.1.7.2 (2020-10-28 14:34:19 -0400)
------------------------------------
* Removing debugging. (VanMuur)
* Added null pointer exception handling to the container. (VanMuur)

v0.1.7.1 (2020-10-24 17:34:03 -0400)
------------------------------------
* Fixed a small bug with the transport flash drive. (VanMuur)

v0.1.7 (2020-10-24 12:42:36 -0400)
----------------------------------
* Working matter container operation. (VanMuur)

v0.1.6.3 (2020-10-23 17:32:36 -0400)
------------------------------------
* Fixing the Holo Sign so that it spans multiple blocks. (VanMuur)

v0.1.6.2 (2020-10-23 16:52:43 -0400)
------------------------------------
* Updating release to v0.1.6.2. (VanMuur)
* Setting up the holo sign to have text spread across the signs if they're next to each other. (VanMuur)
* fixed fusion reactor controller facing east (Force)

v0.1.6.1 (2020-10-18 19:25:33 -0400)
------------------------------------
* Updating version for bug fix. (VanMuur)
* Fixing bug when it came to destFilter being null. (VanMuur)

v0.1.6 (2020-10-18 18:53:54 -0400)
----------------------------------
* Initial commit of network flash drives setting destinations. (VanMuur)

v0.1.5 (2020-10-17 12:15:55 -0400)
----------------------------------
* Fixed animated textures for the decomposer and the recycler. (VanMuur)
* Fixing merge problems. (VanMuur)
* Working copy of the animated matter analyzer. (VanMuur)
* Saving state for the matter analyzer before switching branches. (VanMuur)

v0.1.4.1 (2020-10-17 10:00:34 -0400)
------------------------------------
* Updating version number for release. (VanMuur)
* Fixed the sound muffler's description text, and allowed the sound muffler to be placed in the Dimensional Pylon and Microwave. (VanMuur)
* Updating the gradle wrapper properties file for a new distribution URL. (VanMuur)

v0.1.4 (2020-10-15 20:44:34 -0400)
----------------------------------
* Updated version to 0.1.4 and cleaned up PatternDrive.java file. (VanMuur)
* Fixed pattern drives to show different textures based on how full they are. (VanMuur)
* Saving changes before switching branches. (VanMuur)

v0.1.3.1 (2020-10-12 16:40:56 -0400)
------------------------------------
* Updated version before pushing for build. (VanMuur)
* Fixed muffler sounds for Analyzer, Replicator and Decomposer. (VanMuur)

v0.1.3 (2020-10-12 15:44:06 -0400)
----------------------------------
* Dyeable crates work. Need to implement modded dyes still. (VanMuur)
* Initial commit for getting dying Tritanium Crates working. (VanMuur)

v0.1.2.1 (2020-10-11 19:05:50 -0400)
------------------------------------
* Fixing release workflow. (VanMuur)
* Trying to get workflow to kick off again. (VanMuur)
* Fixed a bug in the github workflow file. (VanMuur)
* Fixed a bunch of things with the muffler upgrade. Works in all machines now and has a pattern. (VanMuur)
* Update en_us.lang (Gandalf360one)

v0.1.2 (2020-10-11 16:51:13 -0400)
----------------------------------
* Fixing version number in build. (VanMuur)
* Fixed version number with build. (VanMuur)
* First edition of the sound muffler. (VanMuur)

v0.1.1 (2020-10-11 14:18:24 -0400)
----------------------------------
* Fixing release version number issue. (VanMuur)
* Attempting to fix release number issue. (VanMuur)
* Fixed workflow version number issue. (VanMuur)
* Fixing reversed sights issue. (VanMuur)
* Trying to fix up the gradle.yml file. (VanMuur)
* Trying to fix up the gradle.yml file. (VanMuur)
* Trying to get autobuilds to build on a tag release. (VanMuur)
* Updating workflow to run on push of tag. (VanMuur)
* Changing to only run on new tag create. (VanMuur)
* Holo sights are now renamed properly, and are working in-game. (VanMuur)
* Holosights are now working. Commit before swapping old holo sights and new ones. (VanMuur)
* Updating tag name for the release. (VanMuur)
* Fixing spelling mistake. (VanMuur)
* Trying to get build release to work. (VanMuur)
* Uploading build artifacts on release. (VanMuur)
* Trying to get a build to release. (VanMuur)
* Fixing gradle.yaml file. (VanMuur)
* Fixed issue in gradle.yaml (VanMuur)

src/main/java/matteroverdrive/Reference.java
/*
 * This file is part of MatterOverdrive: Legacy Edition
 * Copyright (C) 2019, Horizon Studio <contact@hrznstudio.com>, All rights reserved.
 *
 * MatterOverdrive: Legacy Edition is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * MatterOverdrive: Legacy Edition is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Matter Overdrive.  If not, see <http://www.gnu.org/licenses>.
 */

package matteroverdrive;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import matteroverdrive.client.data.Color;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class Reference {
    public static final String MOD_ID = "matteroverdrive";
    public static final String MOD_NAME = "MatterOverdrive: Legacy Edition";
    public static final String VERSION = "@VERSION@";
    public static final String VERSION_DATE = "@DATE@";
    public static final String GUI_FACTORY_CLASS = "matteroverdrive.gui.GuiConfigFactory";
    public static final String DEPENDENCIES = "";
    public static final String CHANNEL_NAME = "mo_channel";
    public static final String CHANNEL_WEAPONS_NAME = CHANNEL_NAME + ":weapons";
    public static final String CHANNEL_GUI_NAME = CHANNEL_NAME + ":gui";
    public static final String VERSIONS_CHECK_URL = "https://api.cfwidget.com/mc-mods/minecraft/229694-matter-overdrive";
    public static final String BETA_CHECK_URL = "http://maven.k-4u.nl/matteroverdrive/MatterOverdrive/maven-metadata.xml";
    public static final String DOWNLOAD_URL = "https://minecraft.curseforge.com/projects/matter-overdrive";
    public static final String PATH_GFX = MOD_ID + ":textures/";
    public static final String PATH_WORLD_TEXTURES = PATH_GFX + "world/";
    public static final String PATH_ARMOR = PATH_GFX + "armor/";
    public static final String PATH_PARTICLE = PATH_GFX + "particle/";
    public static final String PATH_GUI = PATH_GFX + "gui/";
    public static final String PATH_BLOCKS = PATH_GFX + "blocks/";
    public static final String PATH_FX = PATH_GFX + "fx/";
    public static final String PATH_SHADERS = MOD_ID + ":shaders/";
    public static final String PATH_ELEMENTS = PATH_GUI + "elements/";
    public static final String PATH_ICON = PATH_GUI + "icons/";
    public static final String PATH_ENTITIES = PATH_GFX + "entities/";
    public static final String PATH_ITEM = PATH_GFX + "items/";
    public static final String PATH_GUI_ITEM = PATH_GUI + "items/";
    public static final String PATH_MODEL = MOD_ID + ":models/";
    public static final String PATH_MODEL_BLOCKS = PATH_MODEL + "block/";
    public static final String PATH_MODEL_ITEMS = PATH_MODEL + "item/";
    public static final String PATH_SOUNDS = MOD_ID + ":sounds/";
    public static final String PATH_SOUNDS_BLOCKS = PATH_SOUNDS + "blocks/";
    public static final String PATH_INFO = MOD_ID + ":info/";
    public static final String TEXTURE_ARROW_PROGRESS = PATH_ELEMENTS + "progress_arrow_right.png";
    public static final String TEXTURE_FE_METER = PATH_ELEMENTS + "fe.png";
    public static final Color COLOR_WHITE = new Color(255, 255, 255);
    public static final Color COLOR_MATTER = new Color(191, 228, 230);
    public static final Color COLOR_HOLO = new Color(169, 226, 251);
    public static final Color COLOR_YELLOW_STRIPES = new Color(254, 203, 4);
    public static final Color COLOR_HOLO_RED = new Color(230, 80, 20);
    public static final Color COLOR_HOLO_GREEN = new Color(24, 207, 0);
    public static final Color COLOR_HOLO_YELLOW = new Color(252, 223, 116);
    public static final Color COLOR_HOLO_PURPLE = new Color(116, 23, 230);
    public static final Color COLOR_GUI_NORMAL = new Color(62, 81, 84);
    public static final Color COLOR_GUI_LIGHT = new Color(100, 113, 136);
    public static final Color COLOR_GUI_LIGHTER = new Color(139, 126, 168);
    public static final Color COLOR_GUI_DARK = new Color(44, 54, 52);
    public static final Color COLOR_GUI_DARKER = new Color(34, 40, 37);
    public static final Color COLOR_GUI_ENERGY = new Color(224, 0, 0);
    public static final int MODULE_BATTERY = 0;
    public static final int MODULE_COLOR = 1;
    public static final int MODULE_BARREL = 2;
    public static final int MODULE_SIGHTS = 3;
    public static final int MODULE_OTHER = 4;
    public static final int BIONIC_HEAD = 0;
    public static final int BIONIC_ARMS = 1;
    public static final int BIONIC_LEGS = 2;
    public static final int BIONIC_CHEST = 3;
    public static final int BIONIC_OTHER = 4;
    public static final int BIONIC_BATTERY = 5;
    public static final int PACKET_REQUEST_CONNECTION = 0;
    public static final int PACKET_REQUEST_PATTERN_SEARCH = 1;
    public static final int PACKET_REQUEST_NEIGHBOR_CONNECTION = 2;
    public static final int PACKET_REQUEST_VALID_PATTERN_DESTINATION = 3;
    public static final int PACKET_BROADCAST_CONNECTION = 0;
    public static final int PACKET_RESPONCE_ERROR = -1;
    public static final int PACKET_RESPONCE_INVALID = 0;
    public static final int PACKET_RESPONCE_VALID = 1;
    public static final byte MODE_REDSTONE_NONE = 2;
    public static final byte MODE_REDSTONE_HIGH = 1;
    public static final byte MODE_REDSTONE_LOW = 0;
    public static final String MODEL_SPHERE = PATH_MODEL_BLOCKS + "sphere.obj";
    public static final String MODEL_CHARGING_STATION = PATH_MODEL_BLOCKS + "charging_station.obj";
    public static final String MODEL_PATTERN_STORAGE = PATH_MODEL_BLOCKS + "pattern_storage.obj";
    public static final String MODEL_REPLICATOR = PATH_MODEL_BLOCKS + "replicator.obj";
    public static final String MODEL_TRITANIUM_CRATE = PATH_MODEL_BLOCKS + "tritanium_crate.obj";
    public static final String MODEL_INSCRIBER = PATH_MODEL_BLOCKS + "inscriber.obj";
    public static final String CONFIG_KEY_REDSTONE_MODE = "redstoneMode";
    public static final String CHEST_GEN_ANDROID_HOUSE = "android_house";
    public static final String WORLD_DATA_MO_GEN_POSITIONS = "MatterOverdriveWorldGenPositions";
    public static final String UNICODE_LEGENDARY = "\u272a";
    public static final String UNICODE_COMPLETED_OBJECTIVE = "\u25a0";
    public static final String UNICODE_UNCOMPLETED_OBJECTIVE = "\u25a1";
}
